packages/framework/src/client.ts

```diff
@@ -70,10 +70,12 @@ export class Client {
       strictAuthentication: true,
     };
 
-    if (providedOptions?.apiKey !== undefined) {
-      builtConfiguration.apiKey = providedOptions.apiKey;
-    } else if (process.env.NOVU_API_KEY !== undefined) {
-      builtConfiguration.apiKey = process.env.NOVU_API_KEY;
+    builtConfiguration.apiKey = providedOptions?.apiKey || process.env.NOVU_SECRET_KEY || process.env.NOVU_API_KEY;
+
+    if (!builtConfiguration.apiKey) {
+      throw new Error(
+        'Missing secret key. Set the NOVU_SECRET_KEY environment variable or pass an apiKey to the client options.'
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** ‏ 
```suggestion
        'Missing secret key. Set the NOVU_SECRET_KEY environment variable or pass a secret key to the client options.'
```